THREESCALE-7497: Removal of 3scale Istio adapter docs to add 3scale WebAssembly docs. (#36249)
openshift/openshift-docs: modules/ossm-threescale-webassembly-module-configuration.adoc

// Module included in the following assembly:
//
// service_mesh/v2x/ossm-threescale-webassembly-module.adoc

[id="ossm-threescale-webassembly-module-configuration_{context}"]
= The 3scale WebAssembly module configuration

> The `spec.config` of the `ServiceMeshExtension` custom resource is used to provide the configuration that the `Proxy-WASM` module will read from.

Reviewer: ```suggestion
The `ServiceMeshExtension` custom resource spec provides the configuration that the `Proxy-WASM` module reads from.
```
suggestion, future tense

> The `spec.config` configuration is embedded in the host and read by the `Proxy-WASM` module. Typically, the configurations are in the _JSON_ file format for the modules to parse, however the `ServiceMeshExtension` resource will interpret the `spec.config` field value as _YAML_. This then converts the value to _JSON_ for consumption by the module.

Reviewer: ```suggestion
The spec is embedded in the host and read by the `Proxy-WASM` module. Typically, the configurations are in the _JSON_ file format for the modules to parse, however the `ServiceMeshExtension` resource can interpret the spec value as _YAML_ and convert it to _JSON_ for consumption by the module.
```
suggestion

> If you use the `Proxy-WASM` module in standalone mode, you must write the configuration using the _JSON_ format. Using the _JSON_ format means using escaping and quoting where needed within the `host` configuration files, for example `Envoy`. When you use the WebAssembly module with the `ServiceMeshExtension` resource, the configuration is in the _YAML_ format. In this case, an invalid configuration will force the module to show diagnostics based on its _JSON_ representation to a sidecar's logging stream.

Reviewer: ```suggestion
If you use the `Proxy-WASM` module in standalone mode, you must write the configuration using the _JSON_ format. Using the _JSON_ format means using escaping and quoting where needed within the `host` configuration files, for example `Envoy`. When you use the WebAssembly module with the `ServiceMeshExtension` resource, the configuration is in the _YAML_ format. In this case, an invalid configuration forcea the module to show diagnostics based on its _JSON_ representation to a sidecar's logging stream.
```

// Module included in the following assembly:
//
// service_mesh/v2x/ossm-threescale-webassembly-module.adoc

[id="ossm-threescale-webassembly-module-configuration_{context}"]
= The 3scale WebAssembly module configuration

> The `ServiceMeshExtension` custom resource spec provides the configuration that the `Proxy-WASM` module reads from.

Reviewer 2: I notice in this module, Proxy-WASM is marked up with backticks and called a "module". Just checking whether that difference is intentional versus the earlier usage of italics and calling it a "specification".

Author: This is consistent with the upstream documentation. Is this correct @rahulanand16nov ?

Reviewer 3: Both of them are the same thing but written in different styles. To make it more straightforward, we use `Proxy-WASM` spec to create these modules, so to say it in short, those modules are called `Proxy-WASM` modules.

Author: Thanks @rahulanand16nov

> The spec is embedded in the host and read by the `Proxy-WASM` module. Typically, the configurations are in the _JSON_ file format for the modules to parse, however the `ServiceMeshExtension` resource can interpret the spec value as _YAML_ and convert it to _JSON_ for consumption by the module.

Reviewer 2: I don't know that "JSON" or "YAML" need to be italicized or marked up at all (throughout).
